pontoon/base/models.py

```diff
@@ -1762,13 +1772,15 @@ def for_project_locale(self, project, locale, paths=None, statuses=None,
 
         # Filter by search parameters
         if search:
-            search_query = Q(**{'string__icontains': search})
-            search_query |= Q(**{'string_plural__icontains': search})
-            search_query |= Q(**{'translation__string__icontains': search, 'translation__locale': locale})
-            search_query |= Q(**{'comment__icontains': search})
-            search_query |= Q(**{'key__icontains': search})
             # https://docs.djangoproject.com/en/dev/topics/db/queries/#spanning-multi-valued-relationships
-            entities = Entity.objects.filter(search_query, pk__in=entities).distinct()
+            entities = (
+                Entity.objects.filter(
+                    Q(translation__string__icontains=search) | Q(translation__entity_document__icontains=search),
```

Comment:
I probably wanted to avoid redundancy in db -> however I can concatenate these fields... I don't remember any specific edge-case that could prevent that.